task/net/README.md

```diff
@@ -1,6 +1,155 @@
 # About
 The `net` task implements a small netstack based on [_smoltcp_](https://github.com/smoltcp-rs/smoltcp)
 
+# Design of the SP Network Stack
+
+NOTE: This network stack is specialized to the SP's needs and is not, at the
+moment, intended as a fully general network stack.
+
+In particular, we currently only support IP and UDP, with a focus on IPv6.
+
+## The Net Task
+
+The `net` task is responsible for maintaining system network state. It includes:
+
+1. The Ethernet MAC and PHY drivers.
+2. Ethernet DMA support and buffer management.
+3. The IP stack itself (`smoltcp`).
+4. Buffers for each socket defined in the system.
+
+The `net` task itself is designed to manage network interface events as quickly
+as possible, without blocking on other tasks. It sends no IPCs during its normal
+operation.
+
+It exposes an IPC interface that other tasks can use to request services, like
+sending or receiving packets.
+
+## Global configuration
+
+One useful property of our application is that we can predict the full set of
+required network resources -- buffers, sockets, and the like -- at compile time,
+and allocate them statically.
+
+This information is defined in the `app.toml` file's `config.net` section. Here
+is an example of a UDP echo service (port 7) in an early version of the
+netstack:
+
+```toml
+[config.net.sockets.echo]
+kind = "udp"
+owner = {name = "udpecho", notification = 1}
```

Comment:
We now prefer to use named sockets, e.g.
```suggestion
owner = {name = "udpecho", notification = "socket"}
```

It may also be helpful to show the other side, e.g.
```toml
[tasks.udpecho]
name = "task-udpecho"
priority = 5
max-sizes = {flash = 16384, ram = 8192}
stacksize = 4096
start = true
task-slots = ["net"]
notifications = ["socket"]
```

Reply:
Agreed.